src/DotNetty.Codecs/Protobuf/ProtobufVarint32FrameDecoder.cs

```diff
@@ -0,0 +1,135 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+namespace DotNetty.Codecs.Protobuf
+{
+    using System.Collections.Generic;
+    using System.Diagnostics.Contracts;
+    using DotNetty.Buffers;
+    using DotNetty.Transport.Channels;
+
+    ///
+    /// A decoder that splits the received {@link ByteBuf}s dynamically by the
+    /// value of the Google Protocol Buffers
+    /// http://code.google.com/apis/protocolbuffers/docs/encoding.html#varints
+    /// Base 128 Varints integer length field in the message. 
+    /// For example:
+    /// 
+    /// BEFORE DECODE (302 bytes)       AFTER DECODE (300 bytes)
+    /// +--------+---------------+      +---------------+
+    /// | Length | Protobuf Data |----->| Protobuf Data |
+    /// | 0xAC02 |  (300 bytes)  |      |  (300 bytes)  |
+    /// +--------+---------------+      +---------------+
+    ///
+    public class ProtobufVarint32FrameDecoder : ByteToMessageDecoder
```

Comment:
Well, technically variant length is not unique to protobuf. Both proto2 and proto3 use it now.